config/setup_logging.rb

```diff
@@ -92,24 +94,35 @@ def _configure_lograge
     config.lograge.formatter = Lograge::Formatters::Json.new
     config.lograge.base_controller_class = ['ActionController::Base']
     config.lograge.custom_options = ->(event) do
+      request = event.payload[:request]
+      headers = request&.headers
+      headers_env = headers&.env || {}
+
+      remote_ip = request&.remote_ip.presence || event.payload[:remote_ip].presence
+      client_ip = headers_env['HTTP_CLIENT_IP'].presence
+      x_forwarded_for = headers_env['HTTP_X_FORWARDED_FOR'].presence || event.payload[:x_forwarded_for].presence
+      # RFC 7239 dictates the first IP in X-Forwarded-For is the client IP
+      forwarded_ip = x_forwarded_for&.split(',')&.first&.strip.presence
```

Comment:
Given that we're setting these variables here, maybe we should move the logic out of the hash and put all the logic up here.  For example:

```
remote_ip = request&.remote_ip.presence || event.payload[:remote_ip].presence || || forwarded_ip || client_ip || remote_addr
```

which would let us do:
```
{
...
  remote_ip: remote_ip
...
}
```
below.

What do you think?

Reply:
hoisted logic up, removed untrusted vars from remote_ip, added annotations